### PR TITLE
Add core has bower depencies

### PR DESCRIPTION
**Pull request by ndufreche**
Large re-factoring on all the applications and components to remove tb.core and set it has bower dependency.

There is some cleaning to.

GL & HF

PS: There is an undefined require.js bug on the page load but I have try lot functionalities and they still working.

### CODE REVIEW

File: src/require.config.build.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ require.config({
     baseUrl: 'resources/',
     catchError: true,
     paths: {
-        'tb.core': 'src/tb/main.build',
+        'Core': 'src/tb/main.build',
```
Comment (mickaelandrieu) [inline src/require.config.build.js]: are you sure of this path ? I believed you have removed any ``tb`` from the path

File: src/tb/component/cke/main.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*global CKEDITOR:false */
-define(['tb.core.Utils', 'tb.core.ApplicationManager', 'jquery', 'component!rtemanager'], function (Utils, ApplicationManager, jQuery, RteManager) {
```
Comment (mickaelandrieu) [inline src/tb/component/cke/main.js]: is ApplicationManager part of Api now ?

File: src/tb/component/formbuilder/form/element/views/form.element.view.nodeSelector.js [hunk 3]
```diff
@@ -17,7 +17,7 @@
  * along with BackBee. If not, see <http://www.gnu.org/licenses/>.
  */
 
-define(['tb.core', 'tb.core.Renderer', 'tb.core.ApplicationManager', 'BackBone', 'jquery'], function (Core, Renderer, ApplicationManager, Backbone, jQuery) {
+define(['Core', 'Core/Renderer', 'Core/ApplicationManager', 'BackBone', 'jquery'], function (Core, Renderer, ApplicationManager, Backbone, jQuery) {
```
Comment (mickaelandrieu) [inline src/tb/component/formbuilder/form/element/views/form.element.view.nodeSelector.js]: sounds weird to me:
sometimes applicationManager is aliased as "Core" and is 'Core' and sometimes it's aliased as 'ApplicationManager' and is 'Core/ApplicationManager'

File: src/tb/component/searchengine/main.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*global $:false, jQuery:false */
-define(['tb.core', 'require', 'jquery', 'jsclass', 'datetimepicker', 'text!../searchengine/templates/layout.tpl'], function (Core, require, corejQuery) {
+define(['Core', 'require', 'jquery', 'jsclass', 'datetimepicker', 'text!../searchengine/templates/layout.tpl'], function (Core, require, corejQuery) {
```
Comment (mickaelandrieu) [inline src/tb/component/searchengine/main.js]: never see this syntax: 'text!../foo/bar' we need to document the way routing (?) works ?

File: src/require.config.build.js (hunk 1 above)
Comment (ndufreche) [inline src/require.config.build.js]: This is not a good path, but I have to remove the entire file.

File: src/tb/component/cke/main.js (hunk 2 above)
Comment (ndufreche) [inline src/tb/component/cke/main.js]: Yes ApplicationManager is a part the Api.

File: src/tb/component/formbuilder/form/element/views/form.element.view.nodeSelector.js (hunk 3 above)
Comment (ndufreche) [inline src/tb/component/formbuilder/form/element/views/form.element.view.nodeSelector.js]: Yes is a forget.
But ApplicationManager can be used in the 2 modes.

File: src/tb/component/searchengine/main.js (hunk 4 above)
Comment (ndufreche) [inline src/tb/component/searchengine/main.js]: Yes it's a valid syntax, but not the better.

**Comment by mickaelandrieu** [comment]
ping @fkroockmann, is it ok ? can we merge ?

**Comment by ndufreche** [comment]
@backbee/core-team I'have push the next step on BbCoreJs packaging, it's the minification of all the vendors only one file.
```bash
grunt  requirejs
```
Generate a dist/vendor.js file

File: src/tb/apps/content/plugins/contentplugin.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 (function () {
     'use strict';
-    define(['tb.core.Api'], function (Core) {
+    define(['Core/Api'], function (Core) {
```
Comment (fkroockmann) [inline src/tb/apps/content/plugins/contentplugin.js]: Better use only Core

File: src/tb/apps/content/plugins/contentselector/main.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
-define(['content.pluginmanager', 'tb.core.Api', 'component!contentselector', 'jquery', 'content.manager', 'jsclass'], function (PluginManager, Core, ContentSelector, jQuery, ContentManager) {
+define(['content.pluginmanager', 'Core/Api', 'component!contentselector', 'jquery', 'content.manager', 'jsclass'], function (PluginManager, Core, ContentSelector, jQuery, ContentManager) {
```
Comment (fkroockmann) [inline src/tb/apps/content/plugins/contentselector/main.js]: better use only Core

File: src/tb/apps/content/plugins/rte/adapter/rteadapter.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 (function () {
     'use strict';
-    define(['tb.core.Api'], function (Core) {
+    define(['Core/Api'], function (Core) {
```
Comment (fkroockmann) [inline src/tb/apps/content/plugins/rte/adapter/rteadapter.js]: Better use only Core

File: src/tb/apps/content/plugins/rte/main.js [hunk 8]
```diff
@@ -3,7 +3,7 @@ require.config({
         'rteadapter': 'src/tb/apps/content/plugins/rte/adapter/rteadapter' //plugin
     }
 });
-define(['content.pluginmanager', 'component!rtemanager', 'tb.core.Api', 'jquery', 'jsclass'], function (PluginManager, RteManager, Core, jQuery) {
+define(['content.pluginmanager', 'component!rtemanager', 'Core/Api', 'jquery', 'jsclass'], function (PluginManager, RteManager, Core, jQuery) {
```
Comment (fkroockmann) [inline src/tb/apps/content/plugins/rte/main.js]: Better use only Core

File: src/tb/apps/contribution/views/contribution.view.index.js [hunk 9]
```diff
@@ -1,9 +1,9 @@
 define(
     [
-        'tb.core.Api',
+        'Core/Api',
```
Comment (fkroockmann) [inline src/tb/apps/contribution/views/contribution.view.index.js]: Better use only Core

File: src/tb/apps/content/plugins/contentselector/main.js (hunk 6 above)
Comment (fkroockmann) [inline src/tb/apps/content/plugins/contentselector/main.js]: Can you use carriage return for define ?

File: src/tb/apps/layout/controllers/main.controller.js [hunk 10]
```diff
@@ -16,7 +16,7 @@
  * You should have received a copy of the GNU General Public License
  * along with BackBee. If not, see <http://www.gnu.org/licenses/>.
  */
-define(['require', 'tb.core', 'tb.core.Renderer', "component!treeview", 'jquery', 'component!contentselector', 'component!dataview', 'component!datastore'], function (require, bbCore, Renderer, TreeViewMng, jQuery, ContentSelector) {
+define(['require', 'Core', 'Core/Renderer', "component!treeview", 'jquery', 'component!contentselector', 'component!dataview', 'component!datastore'], function (require, bbCore, Renderer, TreeViewMng, jQuery, ContentSelector) {
```
Comment (fkroockmann) [inline src/tb/apps/layout/controllers/main.controller.js]: Can you use carriage return for define ?

File: src/tb/apps/main/main.js [hunk 11]
```diff
@@ -33,7 +33,7 @@ require.config({
     }
 });
 
-define('app.main', ['tb.core', 'tb.core.ApplicationManager', 'main.view.index', 'jquery', 'component!popin', 'bootstrapjs'], function (Core, ApplicationManager, MainViewIndex, jQuery, Popin) {
+define('app.main', ['Core', 'Core/ApplicationManager', 'main.view.index', 'jquery', 'component!popin', 'bootstrapjs'], function (Core, ApplicationManager, MainViewIndex, jQuery, Popin) {
```
Comment (fkroockmann) [inline src/tb/apps/main/main.js]: Can you use carriage return for define ?

File: src/tb/apps/page/main.js [hunk 12]
```diff
@@ -52,7 +52,7 @@ require.config({
     }
 });
 
-define('app.page', ['tb.core.Api', 'page.widget.InformationPage'], function (Core, InformationPage) {
+define('app.page', ['Core/Api', 'page.widget.InformationPage'], function (Core, InformationPage) {
```
Comment (fkroockmann) [inline src/tb/apps/page/main.js]: Better use only Core

File: src/tb/apps/page/views/page.view.clone.js [hunk 13]
```diff
@@ -20,7 +20,7 @@
 define(
     [
         'require',
-        'tb.core.Api',
+        'Core/Api',
```
Comment (fkroockmann) [inline src/tb/apps/page/views/page.view.clone.js]: Better use only Core

File: src/tb/apps/page/views/page.view.tree.js [hunk 14]
```diff
@@ -19,7 +19,7 @@
 
 define(
     [
-        'tb.core.Api',
+        'Core/Api',
```
Comment (fkroockmann) [inline src/tb/apps/page/views/page.view.tree.js]: Better use only Core

File: src/tb/apps/user/controllers/user.controller.js [hunk 15]
```diff
@@ -17,7 +17,7 @@
  * along with BackBuilder5. If not, see <http://www.gnu.org/licenses/>.
  */
 define(
-    ['tb.core', 'tb.core.Renderer', 'user/entity/user', 'component!notify', 'require', 'tb.core.Utils', 'jquery'],
+    ['Core', 'Core/Renderer', 'user/entity/user', 'component!notify', 'require', 'Core/Utils', 'jquery'],
```
Comment (fkroockmann) [inline src/tb/apps/user/controllers/user.controller.js]: Can you use carriage return for define ?

File: src/tb/apps/user/views/popin.view.js [hunk 16]
```diff
@@ -17,7 +17,7 @@
  * along with BackBuilder5. If not, see <http://www.gnu.org/licenses/>.
  */
 define(
-    ['require', 'jquery', 'tb.core', 'user/controllers/dnd.controller', 'component!dnd', 'component!popin', 'component!translator', 'text!user/templates/popin.twig'],
+    ['require', 'jquery', 'Core', 'user/controllers/dnd.controller', 'component!dnd', 'component!popin', 'component!translator', 'text!user/templates/popin.twig'],
```
Comment (fkroockmann) [inline src/tb/apps/user/views/popin.view.js]: Can you use carriage return for define ?

File: src/tb/component/cke/main.js [hunk 17]
```diff
@@ -1,5 +1,5 @@
...
+define(['Core', 'Core/Utils', 'jquery', 'component!rtemanager'], function (Core, Utils, jQuery, RteManager) {
```
Comment (fkroockmann) [inline src/tb/component/cke/main.js]: Can you use carriage return for define ?

File: src/tb/component/contentselector/control/searchengine.control.js [hunk 18]
```diff
@@ -1,4 +1,4 @@
-define(['tb.core', 'require', 'jquery', 'jsclass', 'datetimepicker', 'text!cs-templates/searchengine.tpl'], function (Core, require, jQuery) {
+define(['Core', 'require', 'jquery', 'jsclass', 'datetimepicker', 'text!cs-templates/searchengine.tpl'], function (Core, require, jQuery) {
```
Comment (fkroockmann) [inline src/tb/component/contentselector/control/searchengine.control.js]: Can you use carriage return for define ?

File: src/tb/component/contentselector/helper/content.renderer.js [hunk 19]
```diff
@@ -3,11 +3,11 @@ require.config({
         'content.datastore': 'src/tb/component/contentselector/datastore/content.datastore'
     }
 });
-define(['require', 'content.datastore', 'jquery', 'jsclass', 'nunjucks', 'text!cs-templates/content.list.edit.view.tpl', 'text!cs-templates/content.delete.tpl', 'text!cs-templates/content.grid.view.tpl', 'tb.core.Api', 'text!cs-templates/content.list.view.tpl'], function (require) {
+define(['require', 'content.datastore', 'jquery', 'jsclass', 'nunjucks', 'text!cs-templates/content.list.edit.view.tpl', 'text!cs-templates/content.delete.tpl', 'text!cs-templates/content.grid.view.tpl', 'Core', 'text!cs-templates/content.list.view.tpl'], function (require) {
```
Comment (fkroockmann) [inline src/tb/component/contentselector/helper/content.renderer.js]: Can you use carriage return for define ?

File: src/tb/component/contentselector/main.js [hunk 20]
```diff
@@ -9,7 +9,7 @@ require.config({
         'content.datastore': 'src/tb/component/contentselector/datastore/content.datastore'
     }
 });
-define(['tb.core', 'require', 'jquery', 'text!cs-templates/layout.tpl', 'component!popin', 'underscore', 'BackBone', 'component!rangeselector', 'component!dataview', 'component!mask', 'text!cs-templates/layout.tpl', 'content.renderer', 'cs-control/searchengine.control', 'component!jquery-layout', "component!datastore", "component!treeview", "component!pagination", "node.formater", 'nunjucks', 'content.datastore'], function (Core, require, jQuery, layout, PopInMng) {
+define(['Core', 'require', 'jquery', 'text!cs-templates/layout.tpl', 'component!popin', 'underscore', 'BackBone', 'component!rangeselector', 'component!dataview', 'component!mask', 'text!cs-templates/layout.tpl', 'content.renderer', 'cs-control/searchengine.control', 'component!jquery-layout', "component!datastore", "component!treeview", "component!pagination", "node.formater", 'nunjucks', 'content.datastore'], function (Core, require, jQuery, layout, PopInMng) {
```
Comment (fkroockmann) [inline src/tb/component/contentselector/main.js]: Can you use carriage return for define ?

File: src/tb/component/datastore/DataStore.js [hunk 21]
```diff
@@ -1,7 +1,7 @@
-define(['require', 'jquery', 'BackBone', 'tb.core.Api', 'underscore', 'jsclass', 'tb.core.DriverHandler', 'tb.core.RestDriver'], function (require, jQuery, BackBone, Api, underscore) {
+define(['require', 'jquery', 'BackBone', 'Core', 'underscore', 'jsclass', 'Core/DriverHandler', 'Core/RestDriver'], function (require, jQuery, BackBone, Core, underscore) {
```
Comment (fkroockmann) [inline src/tb/component/datastore/DataStore.js]: Can you use carriage return for define ?

File: src/tb/component/dataview/main.js [hunk 22]
```diff
@@ -4,12 +4,11 @@ require.config({
         'dataStore': 'src/tb/component/dataview/DataStore'
     }
 });
-define(['require', 'tb.core.Api', 'BackBone', 'jquery', 'jsclass', 'tb.core.Api', 'text!dataviewTemplate/layout.tpl'], function (require) {
+define(['require', 'Core', 'BackBone', 'jquery', 'jsclass', 'text!dataviewTemplate/layout.tpl'], function (require) {
```
Comment (fkroockmann) [inline src/tb/component/dataview/main.js]: Can you use carriage return for define ?

File: src/tb/component/formbuilder/form/Form.js [hunk 23]
```diff
@@ -17,7 +17,7 @@
  * along with BackBee. If not, see <http://www.gnu.org/licenses/>.
  */
 
-define('tb.component/formbuilder/form/Form', ['tb.core.Api', 'underscore', 'BackBone', 'jsclass'], function (Core, us, Backbone) {
+define('tb.component/formbuilder/form/Form', ['Core', 'underscore', 'BackBone', 'jsclass'], function (Core, us, Backbone) {
```
Comment (fkroockmann) [inline src/tb/component/formbuilder/form/Form.js]: Can you use carriage return for define ?

File: src/tb/component/formbuilder/form/element/views/form.element.view.checkbox.js [hunk 24]
```diff
@@ -17,7 +17,7 @@
  * along with BackBee. If not, see <http://www.gnu.org/licenses/>.
  */
 
-define(['tb.core', 'tb.core.Renderer', 'BackBone', 'jquery'], function (Core, Renderer, Backbone, jQuery) {
+define(['Core', 'Core/Renderer', 'BackBone', 'jquery'], function (Core, Renderer, Backbone, jQuery) {
```
Comment (fkroockmann) [inline src/tb/component/formbuilder/form/element/views/form.element.view.checkbox.js]: Can you use carriage return for define ?

File: src/tb/component/formbuilder/form/element/views/form.element.view.content.js [hunk 25]
```diff
@@ -17,7 +17,7 @@
  * along with BackBee. If not, see <http://www.gnu.org/licenses/>.
  */
 
-define(['tb.core', 'tb.core.Renderer', 'BackBone', 'component!popin', 'jquery'], function (Core, Renderer, Backbone, PopinManager, jQuery) {
+define(['Core', 'Core/Renderer', 'BackBone', 'component!popin', 'jquery'], function (Core, Renderer, Backbone, PopinManager, jQuery) {
```
Comment (fkroockmann) [inline src/tb/component/formbuilder/form/element/views/form.element.view.content.js]: Can you use carriage return for define ?

File: src/tb/component/formbuilder/form/element/views/form.element.view.datetimepicker.js [hunk 26]
```diff
@@ -17,7 +17,7 @@
  * along with BackBee. If not, see <http://www.gnu.org/licenses/>.
  */
 
-define(['tb.core', 'tb.core.Renderer', 'BackBone', 'jquery'], function (Core, Renderer, Backbone, jQuery) {
+define(['Core', 'Core/Renderer', 'BackBone', 'jquery'], function (Core, Renderer, Backbone, jQuery) {
```
Comment (fkroockmann) [inline src/tb/component/formbuilder/form/element/views/form.element.view.datetimepicker.js]: Can you use carriage return for define ?

File: src/tb/component/formbuilder/form/element/views/form.element.view.file.js [hunk 27]
```diff
@@ -18,7 +18,7 @@
  */
 
 /*global Dropzone */
-define(['tb.core', 'tb.core.Renderer', 'BackBone', 'jquery'], function (Core, Renderer, Backbone, jQuery) {
+define(['Core', 'Core/Renderer', 'BackBone', 'jquery'], function (Core, Renderer, Backbone, jQuery) {
```
Comment (fkroockmann) [inline src/tb/component/formbuilder/form/element/views/form.element.view.file.js]: Can you use carriage return for define ?

File: src/tb/component/medialibrary/helpers/mediaitemrenderer.helper.js [hunk 28]
```diff
@@ -3,7 +3,7 @@ require.config({
         'item.templates': 'src/tb/component/medialibrary/templates/'
     }
 });
-define(['tb.core', 'nunjucks', 'jquery', 'component!popin', 'component!mask', 'jsclass', 'text!item.templates/media.viewmode.tpl', 'text!item.templates/media.deletemode.tpl', 'text!item.templates/media.editmode.tpl'], function (Api, nunjucks, jQuery, PopInManager, MaskManager) {
+define(['Core', 'nunjucks', 'jquery', 'component!popin', 'component!mask', 'jsclass', 'text!item.templates/media.viewmode.tpl', 'text!item.templates/media.deletemode.tpl', 'text!item.templates/media.editmode.tpl'], function (Api, nunjucks, jQuery, PopInManager, MaskManager) {
```
Comment (fkroockmann) [inline src/tb/component/medialibrary/helpers/mediaitemrenderer.helper.js]: Can you use carriage return for define ?

File: src/tb/component/medialibrary/main.js [hunk 29]
```diff
@@ -10,7 +10,40 @@ require.config({
...
+    'mediaFolder.contextMenuHelper',
+    'media.datastore',
+    'component!mask',
+    "jsclass",
+    "component!pagination",
+    "component!notify",
+    "component!jquery-layout"
+], function (
+    require,
+    Api,
```
Comment (fkroockmann) [inline src/tb/component/medialibrary/main.js]: Rename to Core

**Pull request merged**